src/reactviews/pages/PublishProject/components/advancedDeploymentOptionsDrawer.tsx

```diff
@@ -0,0 +1,350 @@
+            if (generalEntries.length > 0) {
+                groups.push({
+                    key: "General",
+                    label: loc.publishProject.generalOptions,
+                    entries: generalEntries,
+                });
+            }
+
+            // Add Ignore Options group
+            if (ignoreEntries.length > 0) {
+                groups.push({
+                    key: "Ignore",
+                    label: loc.publishProject.ignoreOptions,
+                    entries: ignoreEntries,
+                });
+            }
+        }
+
+        // Exclude Object Types group
+        if (state.deploymentOptions.objectTypesDictionary) {
+            const baseExcludedTypes = state.deploymentOptions.excludeObjectTypes?.value || [];
+            const excludeEntries = Object.entries(state.deploymentOptions.objectTypesDictionary)
+                .map(([key, displayName]) => {
+                    const baseExcluded = baseExcludedTypes.some(
+                        (excludedType) => excludedType.toLowerCase() === key.toLowerCase(),
+                    );
+
+                    return {
+                        key,
+                        displayName: displayName || key,
+                        description: "",
+                        value: getCurrentValue(key, baseExcluded),
+                    };
+                })
+                .filter((entry) => entry.displayName)
+                .sort((a, b) => a.displayName.localeCompare(b.displayName));
+
+            if (excludeEntries.length > 0) {
+                groups.push({
+                    key: "Exclude",
+                    label: loc.publishProject.excludeObjectTypes,
+                    entries: excludeEntries,
+                });
+            }
+        }
+
+        return groups;
+    }, [state.deploymentOptions, getCurrentValue, loc]);
+
+    // Options change handler, inserts and removes the changed option in localChanges
+    const handleOptionChange = (optionName: string, checked: boolean) => {
+        setLocalChanges((prev) => {
+            const existingChange = prev.find((change) => change.optionName === optionName);
+            if (existingChange) {
+                return prev.filter((change) => change.optionName !== optionName);
+            } else {
+                return [...prev, { optionName, value: checked }];
+            }
+        });
+    };
+
+    const isResetDisabled = localChanges.length === 0;
+
+    // Options reset handler, clears all local changes (reset to base deployment options)
+    const handleReset = () => {
+        setLocalChanges([]);
+    };
+
+    // Options 'OK' button handler, apply local changes to create updated deployment options
+    const handleOk = () => {
+        if (state.deploymentOptions && localChanges.length > 0) {
+            const updatedOptions = structuredClone(state.deploymentOptions);
+
+            // Apply each local change
+            localChanges.forEach(({ optionName, value }) => {
```

Comment:
there are a lot of if and else here and hard to find the actual logic behind it. maybe just me ... I suggest adding more comment explaining what is really happening